hilti/toolchain/src/compiler/validator.cc

```diff
@@ -177,6 +177,17 @@ struct VisitorPost : visitor::PreOrder, public validator::VisitorMixIn {
         return Nothing();
     }
 
+    // Returns an error if the given type is not named but must be named when used.
+    Result<Nothing> checkIfProperlyNamed(QualifiedType* t) {
```

Comment:
I dislike this function name but apparently I can't think because I can't figure out something better that's short

Reply:
I believe we could simplify the call sites by moving the error reporting into the helper function. How about something like this:

```
void checkDeclarationType(Declaration* d, QualifiedType* t) {
    if ( t->type()->isA<hilti::type::Struct>() || t->type()->isA<hilti::type::Enum>() ||
         t->type()->isA<hilti::type::Union>() ) {
        if ( ! t->type()->typeID() )
            error(fmt("%s types must be named in declarations", t->type()->typeClass()), d);
    }
}
```

And then just call it from the declaration visitors: `checkDeclarationType(n, n->type())`.